library/src/main/java/com/novoda/downloadmanager/NotificationConfig.java

```diff
@@ -0,0 +1,53 @@
+package com.novoda.downloadmanager;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.content.Context;
+import android.os.Build;
+import android.support.v4.app.NotificationCompat;
+
+public class NotificationConfig<T> implements NotificationChannelCreator, NotificationCreator<T> {
```

Comment:
This class replaces the old implementations of `NotificationChannelCreator` and `NotificationCreator`. Passing this many constructor parameters ensures that the clients never have to specify IDs or descriptions more than once, removing a potential source of bugs.

Reply:
Happy to rename / create a builder / anything else here.

Reply:
Now there is only one implementation for both interfaces, what about if we unify them under one interface, make it public and then you make this class private, only a factory could be visible (we do that in other parts of the app) so that only the minimum is visible.

Also I have a question, now the client is forced to create a channel if they only want to provide a custom notification. Can we make it more flexible?

Reply:
The `Config` bothers me a bit because it's a shortcut 😂 but tbh it's just me being a nit 😄